pkg/controller/common/tracing/log.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package tracing
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	pkgerrors "github.com/pkg/errors"
+	"go.elastic.co/apm"
+)
+
+// NewLogAdapter returns an implementation of the log interface expected by the APM agent.
+func NewLogAdapter(log logr.Logger) apm.Logger {
+	return &logAdapter{
+		log: log,
```

Comment:
maybe `log.WithName("tracing")`?

Reply:
I would address that probably in a follow up PR with https://github.com/elastic/cloud-on-k8s/issues/2491

Reply:
OK, but just to be clear I'm just suggesting that the tracer's internal logging should probably have a different logger name associated. Nothing to do with log correlation.

Reply:
I use a package level logger called "tracing" for it at the moment.

Reply:
:man_facepalming: so you do, sorry. I made the comment based on the output in https://github.com/elastic/cloud-on-k8s/pull/2462#discussion_r372928203 (which shows `"log.logger": "manager"`). I suppose that's out of date.